lib/reek/smells/smell_detector.rb

```diff
@@ -103,7 +60,6 @@ def exception?(context)
 
       protected
 
-      # NOTE: Needs to be protected so += works for Ruby < 2.2
```

Comment:
It doesn't need to be protected, or it needs to be protected but for a different reason?